pycqed/measurement/hdf5_data.py

```diff
@@ -124,9 +125,135 @@ def encode_to_utf8(s):
     Required because h5py does not support python3 strings
     '''
     # converts byte type to string because of h5py datasaving
-    if type(s) == str:
+    if isinstance(s, str):
         s = s.encode('utf-8')
     # If it is an array of value decodes individual entries
-    elif type(s) == np.ndarray or list:
+    elif isinstance(s, (np.ndarray, list)):
         s = [s.encode('utf-8') for s in s]
     return s
+
+
+def write_dict_to_hdf5(data_dict: dict, entry_point):
+    """
+    Args:
+        data_dict (dict): dictionary to write to hdf5 file
+        entry_point (hdf5 group.file) : location in the nested hdf5 structure
+            where to write to.
+    """
+    for key, item in data_dict.items():
+        # Basic types
+        if isinstance(item, (str, tuple, float, int, bool,
+                             np.float_, np.int_, np.bool_)):
+            entry_point.attrs[key] = item
+        elif isinstance(item, np.ndarray):
+            entry_point.create_dataset(key, data=item)
+        elif item is None:
+            # as h5py does not support saving None as attribute
+            # I create special string, note that this can create
+            # unexpected behaviour if someone saves a string with this name
+            entry_point.attrs[key] = 'NoneType:__None__'
+
+        elif isinstance(item, dict):
+            entry_point.create_group(key)
+            write_dict_to_hdf5(data_dict=item,
+                               entry_point=entry_point[key])
+        elif isinstance(item, list):
+            if len(item) > 0:
```

Comment:
elif isinstance(item, list):
                if isinstance(item[0], (np.int64, np.float64, str, np.float, float, np.float32,int)):
                    item = np.array(item)
                    h5file[path + key] = item
                else:
                    dict_from_list =dict(zip(range(len(item)),item))
                    dict_from_list['_h5save_type'] = 'list'
                    recursively_save_dict_contents_to_group(h5file, path + key + '/', dict_from_list)

this is just my snippet of code on making a list where recursively_save_dict_contents_to_group is the same as your function